src/Parallel/SimpleActionVisitation.hpp

```diff
@@ -131,5 +131,54 @@ void simple_action_visitor(boost::variant<Variants...>& box, Args&&... args) {
       simple_action_visitor_helper<Invokable, ParallelComponent, Variants>(
           box, &iter, &already_visited, std::forward<Args>(args)...));
 }
+
+template <typename Invokable, typename ParallelComponent, typename... Variants,
+          typename... Args>
+typename Invokable::return_type local_synchronous_action_visitor(
+    boost::variant<Variants...>& box, Args&&... args) {
+  // iter is the current element of the variant in the "for loop"
+  int iter = 0;
+  // already_visited ensures that only one visitor is invoked
+  bool already_visited = false;
+  tmpl::conditional_t<std::is_same_v<typename Invokable::return_type, void>,
+                      NoSuchType, typename Invokable::return_type>
+      return_value{};
```

Comment:
Does this support returning const-references? If so, I suggest to add a test for that, else a note in the docs would be useful on the requirements on the `return_type`.